src/drivers/infiniband/hermon.c

```diff
@@ -857,6 +857,47 @@ hermon_dump_cqctx ( struct hermon *hermon, struct ib_completion_queue *cq ) {
 	return 0;
 }
 
+/**
+ * Dump event queue context (for debugging only)
+ *
+ * @v hermon		Hermon device
+ * @v eq		Event queue
+ * @ret rc		Return status code
+ */
+static __attribute__ (( unused )) int
+hermon_dump_eqctx ( struct hermon *hermon, struct hermon_event_queue *eq ) {
+	struct hermonprm_eqc eqctx;
+	int rc;
+
+	memset ( &eqctx, 0, sizeof ( eqctx ) );
+
+	if ( ( rc = hermon_cmd_query_eq ( hermon, eq->eqn, &eqctx ) ) != 0 ) {
+		DBGC ( hermon, "Hermon %p EQN %#lx QUERY_EQ failed: %s\n",
+		       hermon, eq->eqn, strerror ( rc ) );
+		return rc;
+	}
+
+	if (MLX_GET(&eqctx, consumer_counter) != MLX_GET(&eqctx, producer_counter)) {
+		DBGC ( hermon, "Hermon %p EQN %#lx context:\n", hermon, eq->eqn );
+
+		unsigned int eqe_idx_mask;
+		union hermonprm_event_entry *eqe;
+		for (uint32_t x = 0; x < MLX_GET(&eqctx, producer_counter); x++) {
+			eqe_idx_mask = ( HERMON_NUM_EQES - 1 );
+			eqe = &eq->eqe[x & eqe_idx_mask];
+			DBGC ( hermon, "Hermon %p event sub_type=%d, owner=%d type=%08x",
+			       hermon,
+			       MLX_GET ( &eqe->generic, event_sub_type ),
+			       MLX_GET ( &eqe->generic, owner ),
+			       MLX_GET ( &eqe->generic, event_type ));
+			DBGC_HDA ( hermon, 0, &eqe->generic, 32 );
+		}
+	}
+	/* DBGC_HDA ( hermon, 0, &eqctx, sizeof ( eqctx ) ); */
+
+	return 0;
+}
+
```

Comment:
Replaced with https://github.com/ipxe/ipxe/commit/7d32225b5 which creates `hermon_dump_eqctx()` as a direct analogue of `hermon_dump_cqctx()` and `hermon_dump_qpctx()` and splits out the EQE dumping code to a separate `hermon_dump_eqes()` function.

Reply:
Looks good to me - can we keep the "Hermon init done" debug line below?

Reply:
I reworked the patch with just the "init done" message.